packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -958,6 +958,135 @@ class _DatePickerHeader extends StatelessWidget {
 /// The [builder] parameter can be used to wrap the dialog widget
 /// to add inherited widgets like [Theme].
 ///
+/// ### State Restoration
+///
+/// Using this method will not enable state restoration for the date range picker.
+/// In order to enable state restoration for a date range picker, use
+/// [Navigator.restorablePush] or [Navigator.restorablePushNamed] with
+/// [DateRangePickerDialog].
+///
+/// For more information about state restoration, see [RestorationManager].
+///
+/// {@macro flutter.widgets.RestorationManager}
+///
+/// ```dart imports
+/// import 'package:flutter/material.dart';
+/// ```
+///
+/// ```dart
+///
+/// {@tool sample --template=freeform}
```

Comment:
Maybe its worthwhile to create a special restoration template that can be used here and in some of the other restoration samples to reduce the boilerplate a little.

Reply:
I made the templates and moved the existing sample code here: https://github.com/flutter/flutter/pull/78757